Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -0,0 +1,902 @@
+        auto& mag_gamma_mf = macroscopic_properties->getmag_gamma_mf();
+        // extract material properties
+        Array4<Real> const& mag_Ms_arr = mag_Ms_mf.array(mfi);
+        Array4<Real> const& mag_alpha_arr = mag_alpha_mf.array(mfi);
+        Array4<Real> const& mag_gamma_arr = mag_gamma_mf.array(mfi);
+
+        // extract field data
+        Array4<Real> const &M_xface = Mfield[0]->array(mfi);      // note M_xface include x,y,z components at |_x faces
+        Array4<Real> const &M_yface = Mfield[1]->array(mfi);      // note M_yface include x,y,z components at |_y faces
+        Array4<Real> const &M_zface = Mfield[2]->array(mfi);      // note M_zface include x,y,z components at |_z faces
+        Array4<Real> const &Hx_bias = H_biasfield[0]->array(mfi); // Hx_bias is the x component at |_x faces
+        Array4<Real> const &Hy_bias = H_biasfield[1]->array(mfi); // Hy_bias is the y component at |_y faces
+        Array4<Real> const &Hz_bias = H_biasfield[2]->array(mfi); // Hz_bias is the z component at |_z faces
+        Array4<Real> const &Hx = Hfield[0]->array(mfi);           // Hx is the x component at |_x faces
+        Array4<Real> const &Hy = Hfield[1]->array(mfi);           // Hy is the y component at |_y faces
+        Array4<Real> const &Hz = Hfield[2]->array(mfi);           // Hz is the z component at |_z faces
+        Array4<Real> const &Hx_old = Hfield_old[0]->array(mfi);   // Hx_old is the x component at |_x faces
+        Array4<Real> const &Hy_old = Hfield_old[1]->array(mfi);   // Hy_old is the y component at |_y faces
+        Array4<Real> const &Hz_old = Hfield_old[2]->array(mfi);   // Hz_old is the z component at |_z faces
+
+        // extract field data of a_temp_static and b_temp_static
+        Array4<Real> const &a_temp_static_xface = a_temp_static[0]->array(mfi);
+        Array4<Real> const &a_temp_static_yface = a_temp_static[1]->array(mfi);
+        Array4<Real> const &a_temp_static_zface = a_temp_static[2]->array(mfi);
+        Array4<Real> const &b_temp_static_xface = b_temp_static[0]->array(mfi);
+        Array4<Real> const &b_temp_static_yface = b_temp_static[1]->array(mfi);
+        Array4<Real> const &b_temp_static_zface = b_temp_static[2]->array(mfi);
+
+        // extract stencil coefficients
+        Real const *const AMREX_RESTRICT coefs_x = m_stencil_coefs_x.dataPtr();
+        int const n_coefs_x = m_stencil_coefs_x.size();
+        Real const *const AMREX_RESTRICT coefs_y = m_stencil_coefs_y.dataPtr();
+        int const n_coefs_y = m_stencil_coefs_y.size();
+        Real const *const AMREX_RESTRICT coefs_z = m_stencil_coefs_z.dataPtr();
+        int const n_coefs_z = m_stencil_coefs_z.size();
+
+        // extract tileboxes for which to loop
+        Box const &tbx = mfi.tilebox(Hfield[0]->ixType().toIntVect()); /* just define which grid type */
+        Box const &tby = mfi.tilebox(Hfield[1]->ixType().toIntVect());
+        Box const &tbz = mfi.tilebox(Hfield[2]->ixType().toIntVect());
```

Comment:
```suggestion
        Box const &tbx = mfi.tilebox(Hfield[0]->ixType().toIntVect()); /* just define which grid type */
        Box const &tby = mfi.tilebox(Hfield[1]->ixType().toIntVect());
        Box const &tbz = mfi.tilebox(Hfield[2]->ixType().toIntVect());
```

why is Hfield used here and not Mfield that is used to initialize MFIter?

Reply:
Yes, change these to Mfield (I think you forgot to make the change in your suggestion).

Reply:
haha! yes I did! :) sorry about that!

Reply:
Just for my own curiosity, isn't the tilebox type the same for H and M?

Reply:
it should be since they are built on the same boxarray and distribution map.  It's more a "good practice" thing.

Reply:
I'll accept this change, then manually fix and commit